subliminal/providers/opensubtitlescom.py

```diff
@@ -695,8 +695,12 @@ def query(
                     imdb_match=imdb_match,
                     tmdb_match=tmdb_match,
                 )
-                logger.debug('Found subtitle %r', subtitle)
-                subtitles.append(subtitle)
+
+                # Some criteria are redundant, so skip duplicates
+                unique_ids = [s.id for s in subtitles]
```

Comment:
Use a set for O(1) lookups

```suggestion
                unique_ids = {s.id for s in subtitles}
```